### PR TITLE
[expo-store-review] Fix Android crash in failure path

### PR DESCRIPTION
**Pull request by danmaas**
The failure path in StoreReviewModule.kt calls `promise.reject(null)` which leads to the following crash:

```
Attempt to invoke virtual method 'java.lang.String java.lang.Throwable.getMessage()' on a null object reference
```
Full stack trace at: https://sentry.io/share/issue/74f1816267964b74959ea614f36f2d7c/

I've been seeing a couple of these crashes per day in production. This is in a bare app using Expo SDK 38 with the expo-store-review module cherry-picked up to v2.2.0 to get the recently-added Android implementation.

I believe there's some issue with calling the (overloaded) single-argument form of `Promise.reject()` from Kotlin code with a `null` argument. This patch just switches to the two-argument form where you pass an error code and message.
(see packages/@unimodules/core/android/src/main/java/org/unimodules/core/Promise.java)

### CODE REVIEW

File: packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt [hunk 1]
```diff
@@ -49,7 +49,7 @@ class StoreReviewModule(private val mContext: Context)
           if (task.isSuccessful) {
             promise.resolve(null)
           } else {
-            promise.reject(null)
+            promise.reject("ERR_REVIEW_FLOW_FAILED", "The ReviewManager flow was not successful")
```
Comment (EvanBacon) [inline packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt:52]:
```suggestion
            promise.reject("E_STORE_REVIEW_FAILED", "Android ReviewManager task failed")
```

**Review by EvanBacon — APPROVED** [review]
lgtm, thanks for testing

File: packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt [hunk 2]
```diff
@@ -49,7 +49,7 @@ class StoreReviewModule(private val mContext: Context)
           if (task.isSuccessful) {
             promise.resolve(null)
           } else {
-            promise.reject(null)
+            promise.reject("E_STORE_REVIEW_FAILED", "Android ReviewManager task failed")
```
Comment (sjchmiela) [inline packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt:52]:
```suggestion
            promise.reject("ERR_STORE_REVIEW_FAILED", "Android ReviewManager task failed")
```
From what I can tell looking at recent changes in `expo/expo` this is currently the recommended way of tagging errors, see
https://github.com/expo/expo/blob/bce9476185a05a9077e76eb58ab6f7b2f41d5d64/CHANGELOG.md#L54-L56

**Review by sjchmiela — APPROVED** [review]
Superb! 💯

**Comment by sjchmiela** [comment]
Ah, one last thing — would you care to also update the `expo-store-review/CHANGELOG`? 🙏

**Comment by sjchmiela** [comment]
Thank you! 💟

**Pull request merged**